lib/backend/k8sv2/custom/bgpconfig.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package custom
+
+import (
+	"encoding/json"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/projectcalico/libcalico-go/lib/apiv2"
+)
+
+type BGPConfig struct {
+	metav1.TypeMeta `json:",inline"`
+	Metadata        metav1.ObjectMeta          `json:"metadata"`
+	Spec            apiv2.BGPConfigurationSpec `json:"spec"`
+}
+
+type BGPConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	Metadata        metav1.ListMeta `json:"metadata"`
+	Items           []BGPConfig     `json:"items"`
+}
+
+// Required to satisfy Object interface
```

Comment:
btw, I don't think any of this stuff from here below is required if you use the v2 definitions.  if it seems like it is, can you let me know.

Reply:
I tried to use the apiv2 definitions directly and they didn't seem to work. Probably something I did incorrectly. Let me give it another shot..

Reply:
@robbrockbank Using apiv2 definitions ended up working as expected. There was a small subtle conversion thing that I'd missed and it is all sorted out now.